spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingServiceProperties.java

```diff
@@ -200,4 +200,24 @@ public String getBinder(String channelName) {
 		return bindings.get(channelName).getBinder();
 	}
 
+	/**
+	 * Return configuration properties as Map.
+	 * @return map of channel binding configuration properties.
+	 */
+	public Map<String, Object> asMapProperties() {
+		Map<String, Object> properties = new HashMap<>();
+		properties.put("instanceIndex", String.valueOf(getInstanceIndex()));
+		properties.put("instanceCount", String.valueOf(getInstanceCount()));
+		for (String name : getConsumerProperties().stringPropertyNames()) {
+			properties.put(name, getConsumerProperties());
+		}
+		for (String name : getProducerProperties().stringPropertyNames()) {
+			properties.put(name, getProducerProperties());
+		}
+		for (Map.Entry<String, BindingProperties> entry : getBindings().entrySet()) {
+			properties.put(entry.getKey(), entry.getValue());
+		}
+		return properties;
+	}
+
```

Comment:
Since we only use the producer/consumer property **names**, the possibility of collision is nil (as we don't have any overlapping names). To distinguish these properties, I explicitly added `producer.` and `consumer.` prefix added to the `toString` value. I guess this is ok given we only update the message header. I also updated the test case to reflect this case.